src/Onboarding/useOnboarding.tsx

```diff
@@ -27,59 +69,750 @@ const useOnboarding = (baseUrl?: string) => {
+      })
+    },
+    [
+      checkSolanaStatus,
+      getOnboardingRecord,
+      getBalances,
+      getOraclePrice,
+      solana.status.isHelium,
+      getSolanaAssertData,
+      getHeliumAssertData,
+    ]
+  )
+
+  const submitAssertLocation = useCallback(
+    async ({
+      assertLocationTxn,
+      httpClient,
+      gateway,
+    }: {
+      assertLocationTxn: string
+      httpClient?: Client
+      gateway: string
+    }): Promise<PendingTransaction | undefined> => {
+      checkSolanaStatus()
+
+      const client = httpClient || heliumHttpClient
+
+      let txnStr = assertLocationTxn
+
+      const hotspot = await getHeliumHotspotInfo({
+        hotspotAddress: gateway,
+        httpClient,
+      })
+
+      const onboardingRecord = await getOnboardingRecord(gateway)
+      const isFree = hasFreeAssert({ hotspot, onboardingRecord })
+      if (isFree) {
+        // If L1 is helium and txn is free, must submit to onboard server for payer signature
+        const onboardResponse = await onboardingClient.postPaymentTransaction(
+          gateway,
+          assertLocationTxn.toString()
+        )
+
+        handleError(
+          onboardResponse,
+          `unable to post payment transaction for ${gateway}`
+        )
+
+        if (!onboardResponse?.data?.transaction) {
+          throw new Error('failed to fetch txn from onboarding server')
+        }
+        txnStr = onboardResponse.data?.transaction
+      }
+      return client.transactions.submit(txnStr)
+    },
+    [
+      checkSolanaStatus,
+      getHeliumHotspotInfo,
+      getOnboardingRecord,
+      hasFreeAssert,
+      onboardingClient,
+      handleError,
+    ]
+  )
+
+  const submitTransferHotspot = useCallback(
+    async ({
+      transferHotspotTxn,
+      httpClient,
+    }: {
+      transferHotspotTxn: string
+      httpClient?: Client
+    }): Promise<PendingTransaction | undefined> => {
+      const client = httpClient || heliumHttpClient
+      return client.transactions.submit(transferHotspotTxn)
+    },
+    []
+  )
+
+  const createTransferTransaction = useCallback(
+    async ({
+      hotspotAddress,
+      userAddress,
+      newOwnerAddress,
+      ownerKeypairRaw,
+      httpClient,
+    }: {
+      hotspotAddress: string
+      userAddress: string
+      newOwnerAddress: string
+      ownerKeypairRaw?: SodiumKeyPair
+      httpClient?: Client
+    }): Promise<{
+      transferHotspotTxn?: string | undefined
+      solanaTransactions?: string[] | undefined
+    }> => {
+      checkSolanaStatus()
+
+      const client = httpClient || heliumHttpClient
+
+      if (solana.status.isHelium) {
+        const txn = await Transfer.createTransferTransaction({
+          hotspotAddress,
+          userAddress,
+          newOwnerAddress,
+          client,
+          ownerKeypairRaw,
+        })
+        return { transferHotspotTxn: txn.toString() }
+      }
+
+      // TODO: Add paging
+      const hotspots = await solana.getHotspots({ page: 0 })
```

Comment:
Is there a way to get a hotspot by address?

Reply:
Yes

```javascript
const [key] = await keyToAssetKey(daoKey(HNT_MINT)[0], hotspotAddress)
const keyToAsset = await heliumEntityManagerProgram.account.keyToAssetV0.fetch(key)
const assetId = keyToAsset.asset
const asset = getAsset(url, assetId) // however you're using mpl asset api
```